module/DTO/questionsDTO.js

```diff
@@ -0,0 +1,541 @@
+            } else if (childrenKey === "IsNumber") {
+              const typeofChild =
+                childData?.interactions?.response1?.type?.number;
+              obj[childrenKey] =
+                typeofChild.toLowerCase() === "yes" ? true : false;
+            } else if (childrenKey === "max" || childrenKey === "min") {
+              const slider =
+                type === "slider"
+                  ? childData?.interactions.response1.validation.range[
+                  childrenKey
+                  ]
+                  : type === "date"
+                    ? childData?.interactions.validation[childrenKey]
+                    : "";
+              obj[childrenKey] = slider;
+            }
+          }
+          childQuestion[key] = obj;
+        } else if (key === "payload") {
+          childQuestion[key] = {
+            ...keyData,
+            criteriaId: childData.identifier,
+            responseType: type,
+          };
+        } else if (key === "file") {
+          if (
+            childData[keyData["type"]] &&
+            childData[keyData["type"]].length > 0
+          ) {
+            const require = childData?.interactions?.validation?.required;
+            childQuestion[key] = {
+              ...keyData,
+              required:
+                require === true || require === "Yes" || require === "yes"
+                  ? "Yes"
+                  : "No",
+              type: childData?.evidence?.type,
+            };
+          } else childQuestion[key] = "";
+        } else if (key === "updatedAt" || key === "createdAt") {
+          childQuestion[key] = child[keyData];
+        } else if (key === "dateFormat") {
+          childQuestion[key] =
+            childData?.interactions?.response1?.validation?.pattern || "";
+        } else if (key === "options") {
+          childQuestion[key] =
+            childData?.interactions?.response1?.options || [];
+        } else if (key === "showRemarks" || key === "autoCapture") {
+          const typeofChildren = childData[keyData];
+          childQuestion[key] =
+            key !== "autoCapture"
+              ? typeofChildren === true ||
+                typeofChildren === "Yes" ||
+                typeofChildren === "yes"
+                ? true
+                : false
+              : childData[keyData] || false;
+        } else if (key === "questionNumber" || key === "page") {
+          childQuestion[key] =
+            key === "questionNumber" ? `${index + 1}` : baseAssessment.page;
+        } else {
+          childQuestion[key] = childData[keyData]
+            ? childData[keyData]
+            : child[keyData]
+              ? child[keyData]
+              : "";
+        }
+      }
+      resolve({
+        success: true,
+        message: messageConstants.apiResponses.EVIDENCE_FETCHED,
+        data: {
+          ...childQuestion,
+        },
+      });
+    });
+  }
+
+  /**
+   * Generates matrix questions based on the criteria.
+   *
```

Comment:
@jayesh9747 please correct the function signtaure in entire code which u have witten

Reply:
done